libs/src/sdk/api/albums/AlbumsApi.ts

```diff
@@ -0,0 +1,200 @@
+import type { AuthService, StorageService } from '../../services'
+import type {
+  EntityManagerService,
+  WriteOptions
+} from '../../services/EntityManager/types'
+import type { Configuration } from '../generated/default'
+import { PlaylistsApi } from '../playlists/PlaylistsApi'
+import type {
+  DeleteAlbumRequest,
+  FavoriteAlbumRequest,
+  PublishAlbumRequest,
+  RepostAlbumRequest,
+  UnfavoriteAlbumRequest,
+  UnrepostAlbumRequest,
+  UpdateAlbumRequest,
+  UploadAlbumRequest
+} from './types'
+
+export class AlbumsApi {
+  private readonly playlistsApi: PlaylistsApi
+  constructor(
+    configuration: Configuration,
+    storage: StorageService,
+    entityManager: EntityManagerService,
+    auth: AuthService
+  ) {
+    this.playlistsApi = new PlaylistsApi(
+      configuration,
+      storage,
+      entityManager,
+      auth
+    )
+  }
+
+  // TODO: KJ - Check to see if createAlbum method is needed
+
+  /**
+   * Upload an album
+   * Uploads the specified tracks and combines them into an album
+   */
+  async uploadAlbum(
+    requestParameters: UploadAlbumRequest,
+    writeOptions?: WriteOptions
+  ) {
+    const {
+      metadata,
+      trackMetadatas,
+      trackFiles,
+      onProgress,
+      coverArtFile,
+      userId
+    } = requestParameters
+
+    const playlistRes = await this.playlistsApi.uploadPlaylist(
+      {
+        userId,
+        coverArtFile,
+        onProgress,
+        trackFiles,
+        trackMetadatas,
+        metadata: {
+          description: metadata.description,
```

Comment:
wanted to avoid having the albumName field that is in the metadata get passed bc it would fail the type check. there's probably a smarter way to do this, but this was easy

Reply:
Ah ok, no worries we can leave it as is or you could use the spread operator to do it:

```
const { albumName, ...playlistMetadata } = metadata

uploadPlaylist({
    metadata: {
       ...playlistMetadata,
       playlistName: albumName
    }
})
```

Reply:
🧠